scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/solver/PrioritySolver.java

```diff
@@ -221,7 +225,42 @@ public void initializePlan() throws SimulationFacade.SimulationException {
       simulationFacade.simulateActivities(plan.getActivities());
       plan.getActivities().forEach(activity -> simulationFacade.getActivityDuration(activity)
                                                        .ifPresent(activity::setDuration));
+      final var allGeneratedActivities = simulationFacade.getAllGeneratedActivities(problem.getPlanningHorizon().getEndAerie());
+      processNewGeneratedActivities(allGeneratedActivities);
+    }
+  }
+
+  /**
+   * Filters generated activities and makes sure that simulations are only adding activities and not removing them
+   * @param allNewGeneratedActivities all the generated activities from the last simulation results.
+   */
+  private void processNewGeneratedActivities(Map<ActivityInstance, SchedulingActivityInstanceId> allNewGeneratedActivities) {
+    final var activitiesById = plan.getActivitiesById();
+    final var formattedNewGeneratedActivities = new ArrayList<Pair<ActivityInstance, ActivityInstance>>();
+    allNewGeneratedActivities.entrySet().forEach(entry -> formattedNewGeneratedActivities.add(Pair.of(entry.getKey(), activitiesById.get(entry.getValue()))));
+
+    final var copyOld = new ArrayList<>(this.generatedActivityInstances);
+    final var copyNew = new ArrayList<>(formattedNewGeneratedActivities);
+
+    for(final var pairOld: this.generatedActivityInstances){
+      for (final var pairNew : formattedNewGeneratedActivities){
+        if(pairOld.getLeft().equalsInProperties(pairNew.getLeft()) &&
+           pairNew.getRight().equals(pairOld.getRight())){
+          copyNew.remove(pairNew);
+          copyOld.remove(pairOld);
+          //break at first occurrence. there may be several activities equal in properties.
+          break;
+        }
+      }
+    }
+
+    //copyNew contains only things that are new
+    //copyOld contains only present in old but absent in new
+    if(copyOld.size() != 0){
+      throw new Error("Activities have disappeared from simulation, failing");
```

Comment:
Yes it is related to this code. In the future, the fact that some activities have disappeared should change the execution flow. Maybe some goal becomes unsatisfied and the strategy could be to try processing this failed goal again to see if we can achieve the same level of satisfaction as before.

Reply:
Right - so I would classify this under the same category as resources changing from simulation run to simulation run - it's not an error _unless_ it affects the satisfaction of a previously finished goal. Since we don't error for resources changing from run to run, do you think it might be possible to not error if the decomposition of an activity changes between two runs? Or does that create more problems than it solves?

Reply:
I have commented out these lines and put a TODO to remember for an upcoming change related to this issue.